lib/matplotlib/axes/_axes.py

```diff
@@ -1288,132 +1288,228 @@ def eventplot(self, positions, orientation='horizontal', lineoffsets=1,
-        wonderful ways, including full names (``'green'``), hex
-        strings (``'#008000'``), RGB or RGBA tuples (``(0,1,0,1)``) or
-        grayscale intensities as a string (``'0.8'``).  Of these, the
-        string specifications can be used in place of a ``fmt`` group,
-        but the tuple forms can be used only as ``kwargs``.
+        This interface is suitable for simpler use cases, e.g. when quickly
+        drawing something in an interactive session.
 
-        Line styles and colors are combined in a single format string, as in
-        ``'bo'`` for blue circles.
+        ::
 
-        The *kwargs* can be used to set line properties (any property that has
-        a ``set_*`` method).  You can use this to set a line label (for auto
-        legends), linewidth, antialiasing, marker face color, etc.  Here is an
-        example::
+            plot([x,] y, [fmt,])
+            plot([x,] y, [fmt,], [x2,] y2, [fmt2,], ...)  # multiple data sets
 
-            plot([1,2,3], [1,2,3], 'go-', label='line 1', linewidth=2)
-            plot([1,2,3], [1,4,9], 'rs',  label='line 2')
-            axis([0, 4, 0, 10])
-            legend()
+        You can define basic formatting like color, marker and linestyle using
+        a format string *fmt*. The formats are explained in the *Notes*
+        section below.
 
-        If you make multiple lines with one plot command, the kwargs
-        apply to all those lines, e.g.::
+        Multiple sets of *x*, *y*, *[fmt]* groups can be given. In this case,
+        any additional keyword arguments apply to all data sets.
 
-            plot(x1, y1, x2, y2, antialiased=False)
+        Typical examples::
 
-        Neither line will be antialiased.
+            plot(x, y, 'ro')          # plot x and y using blue circle markers
+            plot(x1, y1, 'g^', x2, y2, 'g-')           # plot two sets of data
 
-        You do not need to use format strings, which are just
-        abbreviations.  All of the line properties can be controlled
-        by keyword arguments.  For example, you can set the color,
-        marker, linestyle, and markercolor with::
 
-            plot(x, y, color='green', linestyle='dashed', marker='o',
-                 markerfacecolor='blue', markersize=12).
+        **3. Labelled data interface**
 
-        See :class:`~matplotlib.lines.Line2D` for details.
+        For objects with labelled data (i.e. data that can be accessed by
+        index ``obj['y']``) you can specify the labels to plot and provide
+        the object in the *data* parameter::
 
-        The kwargs are :class:`~matplotlib.lines.Line2D` properties:
+            plot(ylabel, data=obj)
+            plot(xlabel, ylabel, fmt, data=obj)
 
-        %(Line2D)s
+        All indexable objects are supported. This could e.g. be a `dict`, a
+        `pandas.DataFame` or a structured numpy array.
+
+        .. note::
+            The signature ``plot(xlabel, ylabel, data=obj)`` without a format
```

Comment:
This is not true.  If it is ambiguous it will warn, but 

```python

In [27]: plt.plot('x', 'bob', data={'x': range(5), 'bob': range(5)})
Out[27]: [<matplotlib.lines.Line2D at 0x7f621f4fe898>]
```

works as expected.

Reply:
You're right, but I didn't want to advertise a signature that issues a warning and suggests to change the code.

Reply:
It is only ambiguous when the keys in `data` are also valid 'fmt' strings (and in that case we break the ambiguity in favor of pulling from `data`). 

It's not magic, it's attempts to 'do what I mean' which is worse ;)